src/cuga/backend/agent_spawn/runtime.py

```
@@ -0,0 +1,213 @@
+            },
+        )
+        return answer
+
+    async def execute_async(self, task: str, share_workspace: bool = False) -> str:
+        """Fire-and-forget spawn; return future_id for get_agent_result."""
+        parent_thread_id = self._parent_thread_id()
+        set_session_attribute(parent_thread_id)
+
+        future_id = f"future_{uuid4().hex[:8]}"
+        self._spawn_futures[future_id] = {"status": "running", "result": None, "error": None}
+        asyncio.create_task(self._execute_and_store(future_id, task, share_workspace=share_workspace))
```

Ruff (0.15.20)

[warning] Store a reference to the return value of `asyncio.create_task`

(RUF006)